app/forms/security_event_form.rb

```diff
@@ -98,15 +100,40 @@ def validate_jwt
     errors.add(:jwt, err.message)
   end
 
+  def validate_jti
+    if jti.blank?
+      errors.add(:jti, t('risc.security_event.errors.jti_required'))
+      return
+    end
+
+    return if !user || !service_provider
+
+    return unless record_already_exists?
```

Comment:
is there concern about a potential race condition between the check and insert outside of a transaction? Rails unfortunately doesn't make this easy, so I don't feel super strongly if the risk involved is relatively low.

Reply:
Yes, I definitely had that concern.... but the bummer is that there's no way to for sure check without actually creating it. The validations process in the ActiveModel::Validators doesn't make it easy. So I opted to risk a db race condition to be able to generate better error messages most of the time.

Happy to flip that assumption if you feel strongly

Reply:
Nope, I think it's ok given the likelihood.